client/transport.c

```diff
@@ -113,10 +123,18 @@ static int process_transport_block(struct xdag_block *received_block, void *conn
 
 		case XDAG_MESSAGE_BLOCKS_REPLY:
 		{
-			if(!memcmp(received_block->field[1].hash, reply_id, sizeof(xdag_hash_t))) {
+			if(atomic_compare_exchange_strong_explicit(&reply_id, (uint64_t*)received_block->field[1].hash, reply_id_private, memory_order_relaxed, memory_order_relaxed)) {
```

Comment:
Sorry, here I don't understand only a single thread to sync blocks, why we need atomic? Also, the reply_id is the request id to track request/response, why we need to use reply_id_private as fallback option?

Reply:
just to avoid that malicious peer sends many reply, this way only one reply per answer will be accepted